examples/notebooks/controls_example.ipynb

```diff
@@ -0,0 +1,660 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Controls Example\n",
+    "\n",
+    "If you want to run this notebook on your Raspberry Pi, but don't have Jupyter Notebook, you can install it with `pip`.\n",
+    "\n",
+    "`pip install jupyter`\n",
+    "\n",
+    "Then to run it, simply enter `jupyter-notebook` into the Raspberry Pi terminal\n",
+    "and then navigate the `picamera2/examples/notebooks` folder to find this notebook.\n",
+    "\n",
+    "Note: If running in chromium, you may see better performance by turning off hardware acceleration in the chromium settings.\n",
+    "\n",
+    "***"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[0:18:02.268175580] [2137]  INFO Camera camera_manager.cpp:293 libcamera v0.0.0+3548-a11d63f9\n",
+      "[0:18:02.311149512] [2151]  WARN CameraSensorProperties camera_sensor_properties.cpp:163 No static properties available for 'imx477'\n",
+      "[0:18:02.311188660] [2151]  WARN CameraSensorProperties camera_sensor_properties.cpp:165 Please consider updating the camera sensor properties database\n",
+      "[0:18:02.311207530] [2151] ERROR CameraSensor camera_sensor.cpp:591 'imx477 10-001a': Camera sensor does not support test pattern modes.\n",
+      "[0:18:02.328947861] [2151]  INFO RPI raspberrypi.cpp:1352 Registered camera /base/soc/i2c0mux/i2c@1/imx477@1a to Unicam device /dev/media1 and ISP device /dev/media0\n"
+     ]
+    }
+   ],
+   "source": [
+    "from picamera2.picamera2 import *\n",
+    "import time\n",
+    "\n",
+    "picam2 = Picamera2()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Method #1: Using the set_controls function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "my_controls = {'NoiseReductionMode': 0,\n",
```

Comment:
All the "enum" type controls will want changing to use the Python enums provided by libcamera, e.g. `libcamera.NoiseReductionMode.Off`. I'd like to deprecate the use of "magic ints" where we can.

Reply:
I don't see any problems with sticking keeping integers as another method for denoting a mode, as long as the documentation is robust.  I do think using strings is a better method because it clearly represents the mode while reducing the complexity in entering the value.  Requiring the passing of libcamera enums could be confusing, `libcamera.NoiseReductionMode.HighQuality` is a lot to type for something that could also be entered as a string or integer. It requires users to have some knowledge of libcamera, meaning that they would potentially have to review documentation that is separate from picamera2. This could be mitigated by shortening the libcamera enums in the picamera2 module and by providing the right documentation when picamera2 is released.

I understand that picamera2 and libcamera are tightly coupled, but I see picamera2 being used in more educational settings than libcamera, so having options for both general and advanced users is nice to have.

Reply:
It's fine that "magic ints" work, and we should say that "you can use ints too if you want". I just wouldn't want them to be the first thing that people see, or to become the preferred way of coding. I think both enums and strings are fine, it's just that if I saw a piece of code using ints, I honestly wouldn't have a clue what was going on!